drivers/serial/uart_pl011_ambiq.h

```diff
@@ -168,18 +169,36 @@ static int uart_ambiq_pm_action(const struct device *dev, enum pm_device_action
 }
 #endif /* CONFIG_PM_DEVICE */
 
-#define AMBIQ_PL011_UART_DEFINE(n)                                                                 \
-	PM_DEVICE_DT_INST_DEFINE(n, uart_ambiq_pm_action);                                         \
-	static int pwr_on_ambiq_pl011_uart_##n(void)                                               \
-	{                                                                                          \
-		uint32_t module = (DT_INST_REG_ADDR(n) - UART0_BASE) / (UART1_BASE - UART0_BASE);  \
-		am_hal_pwrctrl_periph_e eUARTPowerModule =                                         \
-			((am_hal_pwrctrl_periph_e)(AM_HAL_PWRCTRL_PERIPH_UART0 + module));         \
-		return am_hal_pwrctrl_periph_enable(eUARTPowerModule);                             \
-	}                                                                                          \
-	static inline int clk_enable_ambiq_pl011_uart_##n(const struct device *dev, uint32_t clk)  \
-	{                                                                                          \
-		return clk_enable_ambiq_uart(dev, clk);                                            \
+#if defined(CONFIG_SOC_SERIES_APOLLO2X)
+#define AMBIQ_PL011_UART_DEFINE(n)								 \
+	static int pwr_on_ambiq_pl011_uart_##n(void)						 \
+	{											 \
+		uint32_t module = (DT_INST_REG_ADDR(n) - UART0_BASE) / (UART1_BASE - UART0_BASE);\
+		uint32_t eUARTPowerModule =							 \
+			((uint32_t)(AM_HAL_PWRCTRL_UART0 << module));				 \
+		am_hal_pwrctrl_periph_enable(eUARTPowerModule);					 \
+		return 0;									 \
+	}											 \
+	static inline int clk_enable_ambiq_pl011_uart_##n(const struct device *dev, uint32_t clk)\
+	{											 \
+		uint32_t module = (DT_INST_REG_ADDR(n) - UART0_BASE) / (UART1_BASE - UART0_BASE);\
+		am_hal_clkgen_uarten_set(module, AM_HAL_CLKGEN_UARTEN_EN);			 \
+		return clk_enable_ambiq_uart(dev, clk);						 \
 	}
+#else
+#define AMBIQ_PL011_UART_DEFINE(n)								 \
```

Comment:
does make sense to keep multiple instance of api. optimize with single instance with reg can be stored on dev config